packages/components/src/Helpers.purs

```diff
@@ -73,6 +77,55 @@ dateTimeFormatter =
     , Placeholder "Z"
     ]
 
+-- Local time aware output formatter for ISO 8601.  Not really useful
+-- for input since it uses a fixed offset.
+localDateTimeFormatter :: Int -> Formatter
+localDateTimeFormatter offset =
+  fromFoldable
+    [ YearFull
+    , Placeholder "-"
+    , MonthTwoDigits
+    , Placeholder "-"
+    , DayOfMonthTwoDigits
+    , Placeholder "T"
+    , Hours24
+    , Placeholder ":"
+    , MinutesTwoDigits
+    , Placeholder ":"
+    , SecondsTwoDigits
+    , Placeholder $ if offset == 0 then "Z"
+                    else if offset < 0
+                         then "-" <> offsetStr (negate offset)
+                         else "+" <> offsetStr offset
+    ]
+    where
+      padLeadingZero n | n < 10 = "0" <> show n
+      padLeadingZero n = show n
+      offsetStr o = padLeadingZero (div o 60) <> ":" <> padLeadingZero (rem 0 60)
+
+-- ISO 8601 parser.  Returns offset but doesn't apply it.
+parseLocalDateTime :: String -> Maybe { offset :: Int, dateTime :: DateTime }
+parseLocalDateTime str = do
```

Comment:
Is there a reason why this isn't just the purs equivalent of 
```js
function parseLocalDateTime(str) {
     const date = new Date(str);
    return date && { offset: date.getTimezoneOffset(), dateTime: date }
}
```

Reply:
ah right, because the offset is the input str's offset instead of the timezone offset. I'd still prefer not implementing a parser since the browser ships with one.

Reply:
It's not going to be any easier to try to understand the whole of JS world's date handling for that.

Reply:
Yeah, misread a bit on what the code was actually doing, my bad.